### PR TITLE
fix: Remember previous scroll position with Go back

### PR DESCRIPTION
**Pull request by Tamaarine**
The change I made just basically added behavior to remember scroll when you click the Go back button in your browser.

Refer to this issue.
https://github.com/filebrowser/filebrowser/issues/1870

I store a large number of pictures in filebrowser and it gets annoying whenever I click the X button on the preview it will scroll me to the top again, and I will have to scroll all the way down again.

This is by no mean a good fix, I had to scramble together to learn a little bit of Vue.js and how it worked and looked at other stackoverflow fixes. I'm sorry if this is scuffed but is just a temp fix.

### CODE REVIEW

**Comment by Alfagun74** [comment]
man its shit but its better than nothing, thanks for your effort homie

**Comment by Tamaarine** [comment]
> man its shit but its better than nothing, thanks for your effort homie

😝

**Comment by niubility000** [comment]
I thhink I have fixed it in my own version, try this: http://xxx.xxx.xxx.xxx (deleted) 
You can download the source code in this website. 

I think we need a public demo website for filebrowser officially.

**Comment by Tamaarine** [comment]
> I thhink I have fixed it in my own version, try this: http://pdj.tpddns.cn:16889/ share:share You can download the source code in this website.
> 
> I think we need a public demo website for filebrowser officially.

Oh wow! This is very good! I will take a look at your fix. Was it a simple fix? Just curious. Thank you for this!

**Comment by niubility000** [comment]
@Tamaarine，
I made some changes in file.vue,  ListingItem.vue, preview.vue, style.css, listing.vue.  It's a little bit complicated. (I just forget how to do it now. )

**Comment by Tamaarine** [comment]
> @Tamaarine， I made some changes in file.vue, ListingItem.vue, preview.vue, style.css, listing.vue. It's a little bit complicated. (I just forget how to do it now. )

Okie thank you! I will take a look at it, appreciate it!!!

**Comment by alanmilinovic** [comment]
Is this a general solution that will cover also cases when you for example close the file after editing?

**Comment by niubility000** [comment]
> 
Yes.

**Comment by niubility000** [comment]
> Is this a general solution that will cover also cases when you for example close the file after editing?

Yes.

**Comment by Alfagun74** [comment]
Man fuck your action bot seriously. Who thought this shit was a good idea?

**Comment by alanmilinovic** [comment]
> Man fuck your action bot seriously. Who thought this shit was a good idea?

Bot is not our friend, he is an enemy.

**Comment by Alfagun74** [comment]
still valid

**Comment by alanmilinovic** [comment]
Waiting for review.

**Comment by alanmilinovic** [comment]
Wow, how much time is needed for so simple code review.

**Comment by Alfagun74** [comment]
Painful... Kills contributors efforts

**Comment by alanmilinovic** [comment]
Will this ever be merged? This with scrolling back on top of the page is very not user friendly. Please merge the change in regards to contributor efforts.

**Comment by Alfagun74** [comment]
He won't give a single flying fuck about his contributors i believe.

**Comment by alanmilinovic** [comment]
There are some minor errors in provided code. I guess @Tamaarine should be able to have a look and fix it.

File: frontend/src/router/index.js [hunk 1]
```diff
@@ -35,6 +35,17 @@ const titles = {
 const router = new Router({
   base: baseURL,
   mode: "history",
+  scrollBehavior (to, from, saved) {
+    if (saved) {
+      return new Promise((resolve, reject) => {
```
Comment (alanmilinovic) [inline frontend/src/router/index.js:40]: `reject` needs to be removed because it is defined but never used.

**Comment by alanmilinovic** [comment]
@Tamaarine are you able to have a look on errors and correct them?

**Comment by sebw** [comment]
Still a problem. When opening a photo and going back to the list, I'm taken back to the top of the file list (applies to #1870).

**Comment by Alfagun74** [comment]
this is a tragedy

**Comment by Alfagun74** [comment]
still valid

**Comment by Alfagun74** [comment]
Still valid

File: frontend/src/router/index.js [hunk 2]
```diff
@@ -35,6 +35,17 @@ const titles = {
...
+        setTimeout(() => {
+          resolve({saved})
+        }, 50)
```
Comment (o1egl) [inline frontend/src/router/index.js:43]: Why timeout is needed here?

Comment (Alfagun74) [reply frontend/src/router/index.js:43]: Dude, just accept the PR already. We've been waiting forever. It's functional.

Comment (o1egl) [reply frontend/src/router/index.js:43]: This is not an answer to the question

Comment (alanmilinovic) [reply frontend/src/router/index.js:43]: @Tamaarine can you please answer. We are waiting for a long time for this feature.

**Comment by alanmilinovic** [comment]
I just cannot believe this feature is not implemented yet, it is making user experience very bad and frustrated for such a small change that needs to be merged. Super annoying.

File: frontend/src/router/index.js (hunk 2 above)
Comment (alanmilinovic) [reply frontend/src/router/index.js:43]: > Dude, just accept the PR already. We've been waiting forever. It's functional.

I agree, just accept the PR, short timeout will not break anything.

**Comment by Alfagun74** [comment]
absolutely unbelievable.

**Comment by alanmilinovic** [comment]
There it goes all in the water.

**Comment by niubility000** [comment]
[2188fixed.zip](https://github.com/filebrowser/filebrowser/files/14296690/2188fixed.zip)
I just figured out a good way to do it. Just freezing the list in the background, like the sidebar and headbar. When you are browsing a file(audio, video, pic), don't let it be destroyed. 
I think it works perfectly. Please see the attachment.
I just forgot how to pull a request, let me try.

**Comment by alanmilinovic** [comment]
> [2188fixed.zip](https://github.com/filebrowser/filebrowser/files/14296690/2188fixed.zip) I just figured out a good way to do it. Just freezing the list in the background, like the sidebar and headbar. When you are browsing a file(audio, video, pic), don't let it be destroyed. I think it works perfectly. Please see the attachment. I just forgot how to pull a request, let me try.

Please try to compile it. I would be very excited to see this fixed.

**Pull request closed without merging**